### DISCUSSION TITLE
Unit testing Insert functions

### DISCUSSION THREAD

**Discussion by Danial968** [aarondl/sqlboiler#1191] [Q&A]
I am trying to write some unit tests for my Insert functions but the created_at and updated_at columns is using time.Now(). How can I override the return value so that it is a static timestamp.

Below is the code from the generated sqlboiler models.
```
	var err error
	currTime := time.Now().In(boil.GetLocation())

	if o.CreatedAt.IsZero() {
		o.CreatedAt = currTime
	}
	if queries.MustTime(o.UpdatedAt).IsZero() {
		queries.SetScanner(&o.UpdatedAt, currTime)
	}
```

**Comment by stephenafamo** [comment]
You can either set the time yourself, or register a `BeforeInsertHook` to set the time.

**Comment by Danial968** [comment]
Thanks for the reply, I would like to set the time myself but I am unsure of how to go about doing it. 

The generated code is using the default time package and I am not sure how I can overwrite that function.

Could you share a snippet of how I can set the time myself.

**Reply by stephenafamo** [reply]
Documentation for hooks is [here](https://github.com/volatiletech/sqlboiler#hooks).

If you want to explicitly use hooks for this, you may disable hooks in the generated code entirely using the flag `--no-hooks`
